### ISSUE TITLE
Ability to display images/pdfs in the dashboard

### ISSUE THREAD

**Issue by akashr050** [apache/incubator-superset#8524]
One of the use-case of the superset dashboards for me is to display images by downloading them from the s3 bucket. The database entry has the s3 location and I would like to download the image and display it. 

It will be great if someone can give me directions on how can I implement this.

**Comment by suddjian** [comment]
If you know your urls in advance, you can add Markdown to a dashboard. But it sounds like you're looking for a way to display an image from your data, and I don't think there's a way to do that right now.

It would be cool to add this. I can think of two implementations that might make sense:
- add jinja templating to the markdown chart-type
- add an option in the table chart-type to treat a column as an image link

Maybe both!

**Comment by akashr050** [comment]
@suddjian Thanks for your comments. I think we can start with the jinja templating options and contribute that feature first. But I am not familiar with the codebase so I would really appreciate if you can guide me wrt this. Even which part of the code / files I should start looking at to add this feature will be helpful.

**Comment by suddjian** [comment]
I'm pretty new to the Superset codebase too. 🙌 The markdown gets parsed at https://github.com/apache/incubator-superset/blob/0.34.0/superset/viz.py#L735, so if we want to add some jinja that might be the easiest place.

The charts themselves live in https://github.com/apache-superset/superset-ui-plugins, but it wasn't always that way, and it looks like the markup chart might have only been partially ported over there. The code powering it seems to mostly live in the superset repo.

**Comment by suddjian** [comment]
@akashr050 I found a way to do this!

The table chart type displays html. For a demo go to sql lab, enter the following, and then click Explore:

```
SELECT
    '<a href="https://github.com/apache/incubator-superset">superset link</a>' as link,
    '<img src="https://media.giphy.com/media/lcySndwSDLxC4eOU86/giphy.gif" alt="panda" />' as img
```

If you have a column of s3 links, you could do
```
SELECT CONCAT('<img src="', s3_img_url, '" alt="s3 image" />') as s3_img_html
```

**Comment by lanyusea** [comment]
@suddjian 
I'm trying to get this work with onmouseover/onmouseout

```
SELECT
'<img src="https://media.giphy.com/media/lcySndwSDLxC4eOU86/giphy.gif" onmouseover="this.src='ttps://media.giphy.com/media/lcySndwSDLxC4eOU86/giphy_new.gif' alt="panda" />' as img
```

the resulting sample is correct, but when exploring there is only the 'img' part of code, the 'onmouseover' got disappeared when I check from the developer console.

any idea how to add this feature?

**Comment by suddjian** [comment]
Javascript is stripped out of the HTML to prevent scripting attacks. There may be a feature flag that disables this protection, but pit is not advised.

**Comment by lanyusea** [comment]
@suddjian  any idea which flag/parameter I should modify? trying to find but no luck.
I deployed in my local PC so the security issue should be ok.

**Comment by Karpisek** [comment]
@lanyusea Maybe "ENABLE_JAVASCRIPT_CONTROLS": True,

**Comment by luutomas** [comment]
@suddjian I just tried it but it does not work. The results just return broken picture. Did you need to set up any other parameters?

**Comment by smetis** [comment]
same here, I got into something that looks like CSP error:
`Refused to load the image 'https://sub.domain.com/imageanalysis/Patttern.jpg' because it violates the following Content Security Policy directive: "img-src 'self' blob: data: https://apachesuperset.gateway.scarf.sh https://static.scarf.sh/".`

I set the CSP on the server that host the image, but still it throw the same error. 
Any help?

thanks!
